### ISSUE TITLE
How to use test_case with async/tokio?

### ISSUE THREAD

**Issue by hoijui** [becheran/ntest#27]
Is it possible, and if so, how, to use `test_case` with an sync function?

I tried this:

```rust
    #[test_case("<http://example.net/>", 1)]
    #[test_case("This is a short link <http://example.net/>", 22)]
    #[tokio::test]
    async fn test_link(input: &str, column: usize) {
        // ...
    }
```

**Comment by zeenix** [comment]
This was very much possible and I was relying on this [until the latest tokio release](https://github.com/dbus2/zbus/issues/225). :(

**Comment by hoijui** [comment]
Hmmm :/
The comment in that issue seems to suggest, that changing the order of the annotations should work, but it seems not to work with the `#[test_case(...]` one.
Do you have an idea what would need to be done?

**Comment by zeenix** [comment]
> The comment in that issue seems to suggest, that changing the order of the annotations should work, 

AFAICT, this isn't true.

> Do you have an idea what would need to be done?

I think either tokio or ntest has to fix this. Given that tokio is the big brother here, I think it'd would be `ntest` that has to change. :shrug:

**Comment by becheran** [comment]
@zeenix it was possible? That is surprising to me, but cool. I never explicitly tried/tested #[test_case()] in combination with `tokio`. I know that `#[timeout()]` worked with tokio and now seems broken, but `#[test_case]` would have worked more by accident. In the code I explicitly expand by adding #[test] to the top of a function.

But maybe the order of macro expansion change in tokio is also the issue here somehow. Though, to be honest I don't really have an idea how I can possibly solve this in ntest. Only thing that comes in my mind is adding some kind of `#[test_case_tokio()` macro which also seems a bit strange since this tiny lib should not directly depend on the tokio library.

**Comment by zeenix** [comment]
> @zeenix it was possible? That is surprising to me, but cool. I never explicitly tried/tested #[test_case()] in combination with `tokio`. I know that `#[timeout()]` worked with tokio and now seems broken, but `#[test_case]` would have worked more by accident. In the code I explicitly expand by adding #[test] to the top of a function.

I've to apologize. It was `timeout` that I've been using in 2 of my projects, not `test_case`.  @Darksonn has generously provided a fix for `timeout` in #28 though so my problem is resolved with that. I hope you can review/merge it soon and release. :)
